src/OrbitBase/PromiseHelpersTest.cpp

```diff
@@ -76,4 +76,128 @@ TEST(GetResultFromFutureAndCallContinuation, CallWithResult) {
   EXPECT_TRUE(called);
 }
 
+TEST(HandleErrorAndSetResultInPromise, SuccessVoidInVoidOut) {
+  ErrorMessageOr<void> in{outcome::success()};
+
+  const auto invocable = []() -> void {};
+
+  Promise<ErrorMessageOr<void>> promise;
+  HandleErrorAndSetResultInPromise<ErrorMessageOr<void>> helper{&promise};
+  helper.Call(invocable, in);
+
+  auto future = promise.GetFuture();
+  ASSERT_TRUE(future.IsFinished());
+  EXPECT_TRUE(future.Get().has_value());
+}
+
+TEST(HandleErrorAndSetResultInPromise, SuccessVoidInIntOut) {
+  ErrorMessageOr<void> in{outcome::success()};
+
+  const auto invocable = []() -> int { return 42; };
+
+  Promise<ErrorMessageOr<int>> promise;
+  HandleErrorAndSetResultInPromise<ErrorMessageOr<int>> helper{&promise};
+  helper.Call(invocable, in);
+
+  auto future = promise.GetFuture();
+  ASSERT_TRUE(future.IsFinished());
+  ASSERT_TRUE(future.Get().has_value());
+  EXPECT_EQ(future.Get().value(), 42);
+}
+
+TEST(HandleErrorAndSetResultInPromise, SuccessIntInVoidOut) {
+  ErrorMessageOr<int> in{42};
+
+  const auto invocable = [](int value) { EXPECT_EQ(value, 42); };
+
+  Promise<ErrorMessageOr<void>> promise;
+  HandleErrorAndSetResultInPromise<ErrorMessageOr<void>> helper{&promise};
+  helper.Call(invocable, in);
+
+  auto future = promise.GetFuture();
+  ASSERT_TRUE(future.IsFinished());
+  EXPECT_TRUE(future.Get().has_value());
+}
+
+TEST(HandleErrorAndSetResultInPromise, SuccessIntInIntOut) {
+  ErrorMessageOr<int> in{42};
+
+  const auto invocable = [](int value) -> int {
+    EXPECT_EQ(value, 42);
```

Comment:
interesting, I remember ASSERTS not compiling when called from a function returning non-void. I take it they fixed this.

Reply:
As far as I know `FAIL` and `ASSERT` won't work, but `EXPECT` is fine.